src/linkify-element.js

```diff
@@ -32,8 +32,11 @@ function tokensToNodes(tokens, opts, doc) {
 
 	for (let i = 0; i < tokens.length; i++) {
 		let token = tokens[i];
+		let validated = options.resolve(opts.validate,
+			token.hasProtocol ? token.hasProtocol() : false,
```

Comment:
I would take out the `hasProtocol` argument, since validation isn't necessarily related to the link protocol. If someone want that behaviour, they would just check the given token string e.g., `type !== 'url' || /^(http|ftp)s?:\/\//.test(text)`